core/bind/core_bind.cpp

```diff
@@ -1819,16 +1819,30 @@ Error _Directory::open(const String& p_path) {
 	return OK;
 }
 
-Error _Directory::list_dir_begin() {
+Error _Directory::list_dir_begin(bool p_skip_navigational, bool p_skip_hidden) {
 
 	ERR_FAIL_COND_V(!d,ERR_UNCONFIGURED);
+
+	_list_skip_navigational = p_skip_navigational;
+	_list_skip_hidden = p_skip_hidden;
+
 	return d->list_dir_begin();
 }
 
 String _Directory::get_next(){
 
 	ERR_FAIL_COND_V(!d,"");
-	return d->get_next();
+
+	String next = d->get_next();
+	while (next != "") {
+		if ((_list_skip_navigational && (next == "." || next == ".."))
```

Comment:
Yeah, that code must've been the result of some work in progress. I'll change it.